internal/server/email/forgotdomain.go

```diff
@@ -1,13 +1,21 @@
 package email
 
 import (
-	"github.com/infrahq/infra/internal/server/models"
+	"sort"
+
+	"github.com/infrahq/infra/internal/server/data"
 )
 
 type ForgottenDomainData struct {
-	Domains []models.ForgottenDomain
+	Organizations []data.ForgottenDomain
 }
 
 func SendForgotDomainsEmail(name, address string, data ForgottenDomainData) error {
+	// TODO: this should probably be done in the caller, but we don't have a reasonable
+	// way to test it in the caller. Once we can test email sends from the API handlers
+	// consider moving this to the caller.
+	sort.Slice(data.Organizations, func(i, j int) bool {
+		return data.Organizations[i].LastSeenAt.After(data.Organizations[j].LastSeenAt)
+	})
```

Comment:
The fix for #3726. I went with LastSeenAt sort order assuming that the most recent one is more likely to be the one they want.